pandas/tests/frame/test_join.py

```diff
@@ -0,0 +1,158 @@
+
+
+class TestDataFrameJoin(TestData):
+
+    def test_join(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
+        result = df1.join(df2)
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, None]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='left')
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, None]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='right')
+        expected = pd.DataFrame({'a': [10, 20, None], 'b': [100, 200, 300]},
+                                index=[1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='inner')
+        expected = pd.DataFrame({'a': [20, 10], 'b': [200, 100]},
+                                index=[2, 1])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='outer')
+        expected = pd.DataFrame({'a': [0, 10, 20, None],
+                                 'b': [None, 100, 200, 300]},
+                                index=[0, 1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+    def test_join_sort(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
+        result = df1.join(df2, sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20], 'b': [None, 100, 200]},
+                                index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='left', sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20], 'b': [None, 100, 200]},
+                                index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='right', sort=True)
+        expected = pd.DataFrame({'a': [10, 20, None], 'b': [100, 200, 300]},
+                                index=[1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+        result = df2.join(df1, how='right', sort=True)
+        expected = pd.DataFrame([[None, 0], [100, 10], [200, 20]],
+                                columns=['b', 'a'], index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='inner', sort=True)
+        expected = pd.DataFrame({'a': [10, 20], 'b': [100, 200]},
+                                index=[1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='outer', sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20, None],
+                                 'b': [None, 100, 200, 300]},
+                                index=[0, 1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+    def test_join_index(self):
```

Comment:
The only new method I have created is `test_join`. All the other methods were previously in `tests/frame/test_misc_api.py`and I just moved them to this file I have created `tests/frame/test_join.py`.

Despite their misleading names, the methods `test_join_index` and a `test_join_index_more` do not directly test any `Index` method, but the `DataFrame.join` functionality and the index and columns in the result DataFrame.